src/main/java/org/osiam/resources/scim/UpdateUser.java

```diff
@@ -0,0 +1,858 @@
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+ * CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+ * TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+package org.osiam.resources.scim;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * Class to create a UpdateUser Object to update a existing User
+ */
+public final class UpdateUser {
+
+    private User user;
+
+    private UpdateUser(Builder builder) {
+        user = builder.updateUser.build();
+    }
+
+    /**
+     * the Scim conform User to be used to update a existing User
+     *
+     * @return User to update
+     */
+    public User getScimConformUpdateUser() {
+        return user;
+    }
+
+    /**
+     * The Builder is used to construct instances of the {@link UpdateUser}
+     */
+    public static class Builder {
+
+        private static final String DELETE = "delete";
+        private String userName;
+        private String nickName;
+        private String externalId;
+        private String locale;
+        private String password;
+        private String preferredLanguage;
+        private String profileUrl;
+        private String timezone;
+        private String title;
+        private Name name;
+        private String userType;
+        private String displayName;
+        private Boolean active;
+        private User.Builder updateUser = null;
+        private Set<String> deleteFields = new HashSet<>();
+        private List<Email> emails = new ArrayList<>();
+        private List<Im> ims = new ArrayList<>();
+        private List<PhoneNumber> phoneNumbers = new ArrayList<>();
+        private List<Address> addresses = new ArrayList<>();
+        private List<Entitlement> entitlements = new ArrayList<>();
+        private List<Photo> photos = new ArrayList<>();
+        private List<Role> roles = new ArrayList<>();
+        private List<X509Certificate> certificates = new ArrayList<>();
+        private Set<Extension> extensions = new HashSet<>();
+
+        public Builder() {
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Avoid unnecessary constructors - the compiler will generate these for you](https://www.codacy.com/app/OSIAM/connector4java/file/1980681120/issues/source?bid=3081107&fileBranchId=3117155#l82)